arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -0,0 +1,562 @@
+    return true;
+  }
+  return false;
+}
+
+/*static*/ bool AqlAnalyzer::normalize_json(const irs::string_ref& args,
+                                                    std::string& out) {
+  auto src = VPackParser::fromJson(args);
+  VPackBuilder builder;
+  if (normalize_slice(src->slice(), builder)) {
+    out = builder.toString();
+    return true;
+  }
+  return false;
+}
+
+/*static*/ irs::analysis::analyzer::ptr AqlAnalyzer::make_vpack(irs::string_ref const& args) {
+  auto const slice = arangodb::iresearch::slice(args);
+  return make_slice(slice);
+}
+
+
+/*static*/ irs::analysis::analyzer::ptr AqlAnalyzer::make_json(irs::string_ref const& args) {
+  auto builder = VPackParser::fromJson(args);
+  return make_slice(builder->slice());
+}
+
+AqlAnalyzer::AqlAnalyzer(Options const& options)
+  : irs::analysis::analyzer(irs::type<AqlAnalyzer>::get()),
+    _options(options),
+    _query(new CalculationQueryContext(arangodb::DatabaseFeature::getCalculationVocbase())),
+    _itemBlockManager(&_resourceMonitor, SerializationFormat::SHADOWROWS), 
+    _engine(0, *_query, _itemBlockManager,
+            SerializationFormat::SHADOWROWS, nullptr) {
+  TRI_ASSERT(validateQuery(_options.queryString,
+                           arangodb::DatabaseFeature::getCalculationVocbase())
+                 .ok());
+}
+
+bool AqlAnalyzer::next() {
+  do {
+    if (_queryResults != nullptr) {
+      while (_queryResults->numRows() > _resultRowIdx) {
+        AqlValue const& value =
+            _queryResults->getValueReference(_resultRowIdx++, _engine.resultRegister());
+        if (value.isString() || (value.isNull(true) && _options.keepNull)) {
+          if (value.isString()) {
+            _term.value = arangodb::iresearch::getBytesRef(value.slice());
+          } else {
+            _term.value = irs::bytes_ref::EMPTY;
+          }
+          _inc.value = _nextIncVal;
+          _nextIncVal = !_options.collapsePositions;
+          return true;
+        }
+      }
+    }
+    if (_executionState == ExecutionState::HASMORE) {
+      _executionState = ExecutionState::DONE;  // set to done to terminate in case of exception
+      _resultRowIdx = 0;
+      _queryResults = nullptr;
+      try {
+        std::tie(_executionState, _queryResults) = _engine.getSome(_options.batchSize);
+        TRI_ASSERT(_executionState != ExecutionState::WAITING);
+      } catch (basics::Exception const& e) {
+        LOG_TOPIC("b0026", ERR, iresearch::TOPIC)
+            << "error executing calculation query: " << e.message()
+            << " AQL query:" << _options.queryString;
+      } catch (std::exception const& e) {
+        LOG_TOPIC("c92eb", ERR, iresearch::TOPIC)
+            << "error executing calculation query: " << e.what()
+            << " AQL query:" << _options.queryString;
+      } catch (...) {
+        LOG_TOPIC("bf89b", ERR, iresearch::TOPIC)
+            << "error executing calculation query"
+            << " AQL query:" << _options.queryString;
```

Comment:
```suggestion
            << " AQL query: " << _options.queryString;
      } catch (std::exception const& e) {
        LOG_TOPIC("c92eb", ERR, iresearch::TOPIC)
            << "error executing calculation query: " << e.what()
            << " AQL query: " << _options.queryString;
      } catch (...) {
        LOG_TOPIC("bf89b", ERR, iresearch::TOPIC)
            << "error executing calculation query"
            << " AQL query: " << _options.queryString;
```

Reply:
Ok, if actual validation happens before and the user gets a proper error returned, then it is fine.